tower-buffer/tests/buffer.rs

```diff
@@ -1,16 +1,11 @@
-extern crate futures;
-extern crate tokio_executor;
-extern crate tower_buffer;
-extern crate tower_mock;
-extern crate tower_service;
-
 use futures::prelude::*;
 use tokio_executor::{SpawnError, TypedExecutor};
-use tower_buffer::*;
-use tower_service::*;
+use tower::{
+    buffer::{error::ServiceError as BufferServiceError, Buffer},
```

Comment:
Why is this being renamed?

Reply:
We had an `error::ServiceError` being used in these tests, and it was ambiguous as to where it was being sourced from. `BufferServiceError` makes it clearer out of context.

Reply:
Can the ambiguity be removed using the module instead? I'm not a huge fan of renaming types as that tends to be more confusing IMO (the type is referenced using a different name).

Reply:
Yep, we can do that!

Reply:
Done.